rules/no-hex-escape.js

```diff
@@ -10,7 +10,9 @@ function checkEscape(context, node, value) {
 		context.report({
 			node,
 			message: 'Use Unicode escapes instead of hexadecimal escapes.',
-			fix: fixer => fixer.replaceTextRange([node.start, node.end], fixedValue)
+			fix: fixer => typeof node.start === 'undefined' ?
+				fixer.replaceText(node, fixedValue) :
+				fixer.replaceTextRange([node.start, node.end], fixedValue)
```

Comment:
```suggestion
			fix: fixer => {
				let {range: [start, end], type, tail} = node;

				if (type === 'TemplateElement') {
					start += 1;
					end -= tail ? 1 : 2;
				}

				return fixer.replaceTextRange([start, end], fixedValue);
			}
```

Reply:
@JounQin Try this, and add a test `` const foo = `\\xb1${foo}` `` , this covers `node.tail = false`

Reply:
Not sure how it works with other parser, seems difference between them

https://github.com/babel/babel/issues/6681

Reply:
It seems much more complicated.

Reply:
I think that the way `TemplateElement` works

Reply:
see https://github.com/eslint/espree/blob/8a67b2f76f497fc71b11b0a71c3659d3cc0019b9/lib/espree.js#L329

Reply:
@fisker I mean it should have been covered by my original fix, isn't it?

Reply:
Not sure, but maybe not . If other parser calculate `range` the eslint ast way, but without `start` this may replacing wrong range, right?

Reply:
OK, changed.